### PR TITLE
Wavefunctions: Tests and docstrings.

### PR DESCRIPTION
**Pull request by amilsted**
Also fixed a stupid bug. Hey, it turns out tests are useful!

### CODE REVIEW

**Comment by Thenerdstation** [comment]
> Hey, it turns out tests are useful!

Who would've guessed.

File: examples/wavefunctions/wavefunctions_test.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      tf.random_normal((phys_dim**2, phys_dim**2), dtype=tf.float64),
+      tf.random_normal((phys_dim**2, phys_dim**2), dtype=tf.float64))
+    h = 0.5 * (h + tf.linalg.adjoint(h))
+    h = tf.reshape(h, (phys_dim, phys_dim, phys_dim, phys_dim))
+    H = [h] * (num_sites - 1)
+
+    norm1 = wavefunctions.inner(psi, psi)
+    en1 = sum(wavefunctions.expval(psi, H[i], i) for i in range(num_sites - 1))
+
+    if graph:
```
Comment (Thenerdstation) [inline examples/wavefunctions/wavefunctions_test.py]: Not a big deal, but if you want to simplify the code, `evolve_trotter_defun = tf.function(evlove_trotter)` should also work.

Comment (amilsted) [reply examples/wavefunctions/wavefunctions_test.py]: I could do that, but it doesn't do the same thing as `evolve_trotter_defun`, which is not itself a defun'd function.

Comment (Thenerdstation) [reply examples/wavefunctions/wavefunctions_test.py]: Nah it's ok. Don't feel the need to make another PR

**Comment by Thenerdstation** [comment]
https://github.com/google/TensorNetwork/issues/40

**Pull request merged**